posthog/migrations/0834_delete_invalid_personalapikeys.py

```diff
@@ -0,0 +1,50 @@
+# Generated by Django 4.2.22 on 2025-08-28 22:37
+import logging
+
+from django.db import migrations, models
+
+logger = logging.getLogger(__name__)
+
+
+def purge_null_secure_values(apps, schema_editor):
+    PersonalAPIKey = apps.get_model("posthog", "PersonalAPIKey")
+
+    base_query = PersonalAPIKey.objects.filter(secure_value__isnull=True)
+
+    # get ids that will be deleted
+    ids = []
+    for pk in base_query.values_list("id", flat=True).iterator(chunk_size=1000):
+        ids.append(pk)
+
+    ids_str = ", ".join(str(i) for i in ids)
+    logger.debug(f"Deleting {len(ids)} rows: {ids_str}")
+
+    deleted, _ = base_query.delete()
+    logger.debug(f"Deleted {deleted} rows.")
+
+
+class Migration(migrations.Migration):
+    atomic = True
+    dependencies = [
+        ("posthog", "0833_experiment_primary_metrics_ordered_uuids_and_more"),
+    ]
+
+    operations = [
+        migrations.RunPython(purge_null_secure_values, reverse_code=migrations.RunPython.noop),
+        # prevent locking for more than 3 sec
+        migrations.RunSQL(
+            "SET LOCAL lock_timeout = '3s'; SET LOCAL statement_timeout = '30s';",
+            migrations.RunSQL.noop,
+        ),
+        # make this non-nullable; ~30k rows in prod-us
+        migrations.AlterField(
+            model_name="personalapikey",
+            name="secure_value",
+            field=models.CharField(
+                unique=True,
+                max_length=300,
+                null=False,
+                editable=False,
+            ),
+        ),
```

Comment:
Our largest `PersonalAPIKey` table is in prod-us with 30k rows. I figured this was small enough to handle a full table lock, but I can switch to a constraint-based approach if not.